diff --git a/zip_file.hpp b/zip_file.hpp
index 89d16f1..23e4fb0 100644
--- a/zip_file.hpp
+++ b/zip_file.hpp
@@ -5120,7 +5120,7 @@ static inline tm safe_localtime(const time_t &t)
 #endif
 }
 
-static inline std::size_t write_callback(void *opaque, std::uint64_t file_ofs, const void *pBuf, std::size_t n)
+static inline std::size_t write_callback(void *opaque, mz_uint64 file_ofs, const void *pBuf, std::size_t n)
 {
     auto buffer = static_cast<std::vector<char> *>(opaque);
     
